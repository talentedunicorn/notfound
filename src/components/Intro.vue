<template>
  <section>
    <h1><span>{{ title }}</span>&#32;<span>{{ subtitle }}</span></h1>
    <router-link class="button" to="/catalog">{{ buttonLabel }}</router-link>
  </section>
</template>

<script>
export default {
  name: 'Intro',
  data () {
    return {
      title: 'notfound',
      subtitle: 'Your friendly 404 generator',
      buttonLabel: 'Get started'
    }
  }
}
</script>

<style scoped>
  section {
    align-self: center;
    text-align: center;
  }

  h1 span {
    display: block;
  }

  h1 span:first-of-type {
    color: var(--gray);
  }

  h1 span:last-of-type {
    text-transform: uppercase;
    letter-spacing: 0.25rem;
    font-size: 0.8rem;
    margin-top: var(--spacing);
  }

  .button {
    margin-top: var(--spacing);
    text-transform: uppercase;
    letter-spacing: 0.125rem;
  }
</style>
