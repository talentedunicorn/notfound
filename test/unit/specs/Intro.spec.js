import {shallowMount} from '@vue/test-utils'
import Intro from '@/components/Intro'

describe('Intro.vue', () => {
  it('should load the correct props', () => {
    const wrapper = shallowMount(Intro, {
      stubs: ['router-link'],
      propsData: {
        title: 'test title'
      }
    })

    expect(wrapper.contains('h1')).toBeTruthy()
    expect(wrapper.props().title).toBe('test title')
  })

  it('should match snapshot', () => {
    const wrapper = shallowMount(Intro, {
      stubs: ['router-link']
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
