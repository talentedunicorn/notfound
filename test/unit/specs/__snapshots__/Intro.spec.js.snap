// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro.vue should match snapshot 1`] = `
<section>
  <h1><span>notfound</span> <span>Your friendly 404 generator</span></h1>
  <!---->
</section>
`;
